tracer/src/Datadog.Trace/AppSec/AppSecRequestContext.cs

```diff
@@ -28,8 +29,17 @@ internal void CloseWebSpan(TraceTagCollection tags, Span span)
         {
             if (_wafSecurityEvents.Count > 0)
             {
-                var triggers = JsonConvert.SerializeObject(_wafSecurityEvents);
-                tags.SetTag(Tags.AppSecJson, "{\"triggers\":" + triggers + "}");
+                // Older version of the Agent doesn't support meta struct
+                // Fallback to the _dd.appsec.json tag
+                if (Security.Instance.IsMetaStructSupported())
+                {
+                    span.SetMetaStruct(AppsecKey, MetaStructHelper.ObjectToByteArray(new Dictionary<string, List<object>> { { "triggers", _wafSecurityEvents } }));
```

Comment:
Probably not something you want to do in this PR, but we should probably try to move this MessagePack/JSON serialization into the background thread we already use to serialize spans, instead of blocking the web request.

Reply:
Yep that's will be even better. It will be part of another PR 😄